tests/pylint_plugin_disable.py

```diff
@@ -0,0 +1,47 @@
+"""
+Disable certain checks for the tests.
+
+To disable new checks, add to the `SUPPRESS_CHECKS` dictionary,
+with `message_id` of the check to disable as a key and a list of
+methods that check for that particular message.
+"""
+import os.path
+from typing import TYPE_CHECKING
+
+from pylint.checkers.base import NameChecker
+from pylint.checkers.classes import ClassChecker
+
+if TYPE_CHECKING:
+    from astroid import node_classes, scoped_nodes  # noqa
+    from pylint.lint import PyLinter
+
+
+TESTS_FOLDER = os.path.join(os.path.abspath(os.path.dirname(__file__)), "")
+SUPPRESS_CHECKS = {
+    "protected-access": [
+        ClassChecker.visit_assign,
+        ClassChecker.visit_attribute,
+    ],
+    "blacklisted-name": [
+        NameChecker.visit_global,
+        NameChecker.visit_assignname,
+    ],
+}
+
+
+def is_node_in_tests(node: "node_classes.NodeNG"):
+    module: "scoped_nodes.Module" = node.root()
+    return module.file.startswith(TESTS_FOLDER)
```

Comment:
@efiop, @pmrowla, @pared, I think we could do a similar prefix-based solution to reduce some `relpath`/`abspath` calls.